src/Lock.php

```diff
@@ -21,8 +24,7 @@ class Lock
      * Creates a new lock permit object.
      *
      * @param int $id The lock identifier.
-     * @param callable $releaser A function to be called upon release. The function will be passed this object as the
-     *     first parameter.
+     * @param callable(self): Promise<void> $releaser A function to be called upon release.
      */
     public function __construct(int $id, callable $releaser)
```

Comment:
I've found the `$id` property to be completely pointless.

Reply:
It's used here: https://github.com/amphp/sync/blob/b33f8b7e3b463cd864ed75f4759ad5d74436e632/src/SemaphoreMutex.php#L27

Reply:
I suppose that's helpful. I've otherwise never needed to use the ID.

Reply:
All the semaphores seems to use it to some extent. For instance https://github.com/amphp/sync/blob/b33f8b7e3b463cd864ed75f4759ad5d74436e632/src/LocalSemaphore.php#L30-L51

Reply:
I guess my main point is that I've never used the ID from the consumer side of `Lock`. Here it's just an implementation detail in `LocalSemaphore`. So I guess it's not "pointless," but I haven't found it particularly useful except in the case @kelunik pointed out. Turns out we do use it, so never mind, leave it as-is.

Reply:
@trowski It's hidden from the user anyway... The user never actually creates a lock, he just gets an instance from a mutex or semaphore so he doesn't need to care about the id in __construct at all..